src/App.tsx

```diff
@@ -212,13 +212,9 @@ function App() {
                 <ValueEditor
                   key={`val${index}`}
                   value={value}
-                  index={index + 1}
+                  index={index}
                   onAdd={() => {
-                    setValues((values) => {
-                      const _values = [...values]
-                      _values.splice(index + 1, 0, {value: ''})
-                      return _values
-                    })
+                    setValues((values) => [...values, {value: ''}])
```

Comment:
I've refactored this since it's much simpler to use the spread, as you were already doing, and just adding a new element to the array without using splice. 

No mutation involved in both ways, so this is still ok.